src/Service/Core.php

```diff
@@ -95,58 +96,32 @@ public function createNodes($module) {
     );
   }
 
-  /**
-   * Get initial config from the module's JSON file.
-   *
-   * @param string $module
-   *   The machine name of the module.
-   *
-   * @return array
-   *   Array representation of the configuration settings.
-   */
-  public function getConfigFromJson($module) {
-    $json = [];
-
-    // Get the JSON file for the module.
-    $filename = DRUPAL_ROOT . '/' . drupal_get_path('module', $module) . "/$module.config.json";
-
-    if (file_exists($filename)) {
-      $string = file_get_contents($filename);
-      $json = json_decode($string);
-    }
-
-    return $json;
-  }
-
   /**
    * Get the current text and configuration settings for an app.
    *
    * @param string $module
-   *   The machine name of the extending module.
+   *   The machine name of the module being installed.
    * @param string $language
    *   The language code.
    *
-   * @return string
-   *   The text and configuration settings for the app, as JSON.
-   *
-   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
-   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @return array
+   *   The text and configuration settings for the app json endpoint, as array.
    */
   public function getAppConfig($module, $language) {
-
-    $config = new \StdClass();
-
     // Get the relevant node for the app.
     $node = $this->getAppNode($module, $language);
-    if (!empty($node)) {
-      $app_config = $node->field_spalp_app_config->value;
-      $app_text = $node->field_spalp_app_text->value;
 
-      $config->appConfig = json_decode($app_config);
-      $config->appText = json_decode($app_text);
-    }
+    $config_json = !empty($node->field_spalp_config_json->value) ? $node->field_spalp_config_json->value : NULL;
+
+    $config = json_decode($config_json, TRUE);
```

Comment:
should be using Json::decode
See: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Serialization%21Json.php/function/Json%3A%3Adecode/8.2.x

Reply:
Will implement this with DI implementation